src/Illuminate/Support/Arr.php

```diff
@@ -203,6 +203,12 @@ public static function forget(&$array, $keys)
             while (count($parts) > 1) {
                 $part = array_shift($parts);
 
+                if ($part === '*') {
+                    foreach ($array as &$item) {
+                        static::forget($item, implode('.', $parts));
+                    }
+                }
```

Comment:
Maybe we should go to next key, taking into consideration that we passed all the items wildcard.

```php
    //...
        continue 2;    
    } elseif (isset($array[$part]) && is_array($array[$part])) {
        $array = &$array[$part];
    //...
```